rp2040-hal/src/uart/reader.rs

```diff
@@ -133,19 +133,17 @@ pub(crate) fn read_raw<'b, D: UartDevice>(
 
             let read = device.uartdr.read();
 
+            // If multiple status bits are set, report
+            // the most serious or most specific condition,
+            // in the following order of precedence:
+            // overrun > break > parity > framing
             if read.oe().bit_is_set() {
                 error = Some(ReadErrorType::Overrun);
-            }
-
-            if read.be().bit_is_set() {
+            } else if read.be().bit_is_set() {
                 error = Some(ReadErrorType::Break);
-            }
-
-            if read.pe().bit_is_set() {
+            } else if read.pe().bit_is_set() {
                 error = Some(ReadErrorType::Parity);
-            }
-
-            if read.fe().bit_is_set() {
+            } else if read.fe().bit_is_set() {
```

Comment:
Is the parity bit relevant if there's a framing error?

Reply:
I didn't find error handling code in the C sdk. If both FE and PE are set, embassy reports the parity error, most (but not all) linux serial drivers I checked as well. So there doesn't seem to be a consensus on how to handle that case. And honestly, I think it doesn't matter much.

If there are random bit errors, PE will happen more often than FE. If parity is misconfigured, there will only PE errors (and each character will trigger it). If no parity bit is configured, PE is irrelevant. In general, both errors indicate that the character is probably damaged, so error handling for both is probably identical anyway.